From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Paolo Sabatino <paolo.sabatino@gmail.com>
Date: Wed, 20 Aug 2025 21:40:48 +0200
Subject: [ARCHEOLOGY] workaround for rockchip gigabit ethernet not able to
 init dma

> X-Git-Archeology: - Revision ae11ad5db511f41448066fd29c1c9f5e661b023c: https://github.com/armbian/build/commit/ae11ad5db511f41448066fd29c1c9f5e661b023c
> X-Git-Archeology:   Date: Wed, 20 Aug 2025 21:40:48 +0200
> X-Git-Archeology:   From: Paolo Sabatino <paolo.sabatino@gmail.com>
> X-Git-Archeology:   Subject: workaround for rockchip gigabit ethernet not able to init dma
> X-Git-Archeology:
> X-Git-Archeology: - Revision 29317c6f7e33f2cc509acc0da23b615a7d9d8c31: https://github.com/armbian/build/commit/29317c6f7e33f2cc509acc0da23b615a7d9d8c31
> X-Git-Archeology:   Date: Thu, 18 Sep 2025 22:48:06 +0200
> X-Git-Archeology:   From: Paolo Sabatino <paolo.sabatino@gmail.com>
> X-Git-Archeology:   Subject: update rockchip 32 bit edge kernel to 6.17
> X-Git-Archeology:
---
 drivers/net/ethernet/stmicro/stmmac/stmmac_main.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c b/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
index 111111111111..222222222222 100644
--- a/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
+++ b/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
@@ -3134,8 +3134,8 @@ static int stmmac_init_dma_engine(struct stmmac_priv *priv)
 
 	ret = stmmac_reset(priv, priv->ioaddr);
 	if (ret) {
-		netdev_err(priv->dev, "Failed to reset the dma\n");
-		return ret;
+		netdev_warn(priv->dev, "Failed to reset the dma, device will work with reduced throughput\n");
+		ret = 0;
 	}
 
 	/* DMA Configuration */
-- 
Armbian

